jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -107,7 +106,8 @@ object XGBoost extends Serializable {
       removeMissingValues(verifyMissingSetting(xgbLabelPoints, missing),
         missing, (v: Float) => v != missing)
     } else {
-      removeMissingValues(xgbLabelPoints, missing, (v: Float) => !v.isNaN)
+      removeMissingValues(verifyMissingSetting(xgbLabelPoints, missing),
```

Comment:
Is this change correct? In the `else` clause `missing.isNaN` is true, yet `verifyMissingSetting()` would throw an exception. Seems like the `else` clause would always throw an exception. Is that expected?

Reply:
yes, it's expected 

if you have NaN as missing value, that means you think 0 is a meaningful value in your dataset, however, vectorAssembler has produced a sparsevector for some of your rows (by filtering 0, since vectorassembler only thinks 0 as missing value)

instead of proceeding with this incompatible issue between XGBoost/VectorAssembler, we have to stop the training to avoid messing up the accuracy

else clause does not always throw exception, because when you have every row as dense vector, it would exceed (because we didn't find any 0 filtered by vector assembler...)